Inheritance from models fields

I've got the following snippet
```python
from django.db import models

class PhoneField(models.CharField):
    """Phone field based on CharField with validation."""

    def __init__(self, *args, **kwargs) -> None:
        """Init."""
        options = {
            "max_length": 30,
            "validators": (must_be_valid_number, ),
        }
        options.update(kwargs)
        super().__init__(*args, **options)
```

And there are two problems:
* First - `models.py:3 error: Missing type parameters for generic type "CharField"`

* And second
```
models.py:13: error: Argument 2 to "__init__" of "CharField" has incompatible type "**Dict[str, object]"; expected "Union[str, bytes, None]"
models.py:13: error: Argument 2 to "__init__" of "CharField" has incompatible type "**Dict[str, object]"; expected "Optional[str]"
models.py:13: error: Argument 2 to "__init__" of "CharField" has incompatible type "**Dict[str, object]"; expected "bool"
models.py:13: error: Argument 2 to "__init__" of "CharField" has incompatible type "**Dict[str, object]"; expected "Optional[int]"
models.py:13: error: Argument 2 to "__init__" of "CharField" has incompatible type "**Dict[str, object]"; expected "Optional[Iterable[Union[Tuple[Any, Any], Tuple[str, Iterable[Tuple[Any, Any]]]]]]"
models.py:13: error: Argument 2 to "__init__" of "CharField" has incompatible type "**Dict[str, object]"; expected "str"
models.py:13: error: Argument 2 to "__init__" of "CharField" has incompatible type "**Dict[str, object]"; expected "Iterable[Callable[..., None]]"
models.py:13: error: Argument 2 to "__init__" of "CharField" has incompatible type "**Dict[str, object]"; expected "Optional[Dict[str, Any]]"
```


How can I annotate generic CharField?

And how can I pass kwargs to super?

I did and `incompatible type "**Dict[str, object]"` errors are gone.

But I still get `models.py:3 error: Missing type parameters for generic type "CharField"` error.

--- a/README.md
+++ b/README.md
@@ -347,8 +347,8 @@ def assert_zero_count(model_type: type[models.Model]) -> None:
 
 
 Django `models.Field` (and subclasses) are generic types with two parameters:
-- `_GT`: type that will be returned when getting a value
 - `_ST`: type that can be used when setting a value
+- `_GT`: type that will be returned when getting a value
 
 When you create a subclass, you have two options depending on how strict you want
 the type to be for consumers of your custom field.

--- a/README.md
+++ b/README.md
@@ -353,10 +353,11 @@ Django `models.Field` (and subclasses) are generic types with two parameters:
 When you create a subclass, you have two options depending on how strict you want
 the type to be for consumers of your custom field.
 
-1. Generic subclass
+1. Generic subclass:
+
 ```python
-from django.db import models
 from typing import TypeVar, reveal_type
+from django.db import models
 
 _ST = TypeVar("_ST", contravariant=True)
 _GT = TypeVar("_GT", covariant=True)
@@ -372,10 +373,11 @@ reveal_type(User().my_field) # N: Revealed type is "int"
 User().my_field = "12"  # OK (because Django IntegerField allows str and will try to coerce it)
 ```
 
-2. Non-generic subclass (more strict)
+2. Non-generic subclass (more strict):
+
 ```python
-from django.db import models
 from typing import reveal_type
+from django.db import models
 
 # This is a non-generic subclass being very explicit
 # that it expects only int when setting values.
@@ -390,7 +392,7 @@ reveal_type(User().my_field) # N: Revealed type is "int"
 User().my_field = "12" # E: Incompatible types in assignment (expression has type "str", variable has type "int")
 ```
 
-See mypy section on [generic classes subclasses.](https://mypy.readthedocs.io/en/stable/generics.html#defining-subclasses-of-generic-classes)
+See mypy section on [generic classes subclasses](https://mypy.readthedocs.io/en/stable/generics.html#defining-subclasses-of-generic-classes).
 
 ## Related projects
 

--- a/README.md
+++ b/README.md
@@ -342,6 +342,10 @@ def assert_zero_count(model_type: type[models.Model]) -> None:
 
 ### How to type a custom `models.Field`?
 
+> [!NOTE]
+> This require type generic support, see <a href="#i-cannot-use-queryset-or-manager-with-type-annotations">this section</a> to enable it.
+
+
 Django `models.Field` (and subclasses) are generic types with two parameters:
 - `_GT`: type that will be returned when getting a value
 - `_ST`: type that can be used when setting a value

--- a/README.md
+++ b/README.md
@@ -340,6 +340,54 @@ def assert_zero_count(model_type: type[models.Model]) -> None:
 ```
 
 
+### How to type a custom `models.Field`?
+
+Django `models.Field` (and subclasses) are generic types with two parameters:
+- `_GT`: type that will be returned when getting a value
+- `_ST`: type that can be used when setting a value
+
+When you create a subclass, you have two options depending on how strict you want
+the type to be for consumers of your custom field.
+
+1. Generic subclass
+```python
+from django.db import models
+from typing import TypeVar, reveal_type
+
+_ST = TypeVar("_ST", contravariant=True)
+_GT = TypeVar("_GT", covariant=True)
+
+class MyIntegerField(models.IntegerField[_ST, _GT]):
+    ...
+
+class User(models.Model):
+    my_field = MyIntegerField()
+
+
+reveal_type(User().my_field) # N: Revealed type is "int"
+User().my_field = "12"  # OK (because Django IntegerField allows str and will try to coerce it)
+```
+
+2. Non-generic subclass (more strict)
+```python
+from django.db import models
+from typing import reveal_type
+
+# This is a non-generic subclass being very explicit
+# that it expects only int when setting values.
+class MyStrictIntegerField(models.IntegerField[int, int]):
+    ...
+
+class User(models.Model):
+    my_field = MyStrictIntegerField()
+
+
+reveal_type(User().my_field) # N: Revealed type is "int"
+User().my_field = "12" # E: Incompatible types in assignment (expression has type "str", variable has type "int")
+```
+
+See mypy section on [generic classes subclasses.](https://mypy.readthedocs.io/en/stable/generics.html#defining-subclasses-of-generic-classes)
+
 ## Related projects
 
 - [`awesome-python-typing`](https://github.com/typeddjango/awesome-python-typing) - Awesome list of all typing-related things in Python.

